raiden/blockchain_events_handler.py

```diff
@@ -293,11 +283,22 @@ def handle_secret_revealed(raiden, event, current_block_number):
 
 
 def on_blockchain_event(raiden, event, current_block_number, chain_id):
-    log.debug('BLOCKCHAIN EVENT', node=pex(raiden.address), chain_event=event)
-
     event = decode_event_to_internal(event)
     data = event.event_data
 
+    # filter out events in channels we're not participating in
+    if data['event'] != ChannelEvent.OPENED and data.get('channel_identifier'):
```

Comment:
This should be done at all. What problem is this fixing?

Reply:
#2174 
By registering a single token network event listener/filter with `None` as topics, we get every event on the TokenNetwork. `OPENED` events must be handled regardless of the channel existence, as new channels with us will be added on handling of it, and channels we don't participate becomes new routes. All the other events **which have a `channel_identifier` argument** are the ones previously handled by the `payment_channel` filters, which registered a `[None, [channel_id]]` topic filter (which handled the events with `channel_identifier`/first argument as the `channel_identifier`s we care about aka we participate in. As all events are handled here, I kept a single, `[None]` event filter for the `TokenNetwork`, and filter out the channel specific events here.

Reply:
From #2174 second point you wrote:
> The state machine/**blockchain event handler** must ignore events for channels which it's not a participant

I'd argue I'm not handling the event, I'm preventing events we're not **supposed** to handle at all from being handled. Removing the payment channels filter to stay with the token network one broadened our view of the events. This filter here brings it back. I don't want to spam log events we're not supposed to even look at.

Reply:
I'm arguing this is the wrong place to do the  check, not that it's not required.

Reply:
Right below it are the `handle_` functions. For me, if the event reach that point, it's **already** being handled, and was even logged, and I want to avoid that, as it's a filter for events we **must not** handle at all, therefore, the filtering should be done **above** `handle_` calls, which is either here or in the `RaidenService._callback_new_block`

Reply:
From my understanding this will break the building of the network graph. In `handle_channel_new` we assume that we get all `ChannelOpened` events and handle whether the node is a participant or not there.

Reply:
Ok, that makes sense, but we need to notice this changes the current approach. Currently, the payment channels filters are added only for the channels we participate in. My current PR was intended to only translate the previous behavior and fix #2174 , we're expanding it then. I'll implement.